test/sets.jl

```diff
@@ -1,5 +1,10 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+# OffsetVector is used to test `unique!` on non-standard indices (PR #20619).
```

Comment:
I'm not sure we really care about where it's used. If another tests starts using `OffsetArray`, this comment will be kind of misleading. Also we don't put references to PRs in the code, the git history is enough.

On the line below, you can also break after `||`, which might be more logical.